rust-lisa/src/main/java/it/unipr/cfg/RustCFG.java

```diff
@@ -0,0 +1,193 @@
+			if (getAllExitpoints().isEmpty()) {
+				Set<Statement> exitNodes = nodes
+						.stream()
+						.filter(n -> getAdjacencyMatrix().followersOf(n).isEmpty())
+						.collect(Collectors.toSet());
+
+				for (Statement exit : exitNodes) {
+					addNode(ret);
+					addEdge(new SequentialEdge(exit, ret));
+				}
+			} else
+				addNode(ret);
+
+			// Substitute return with ret nodes
+			for (Statement node : nodes) {
+				if (node instanceof RustReturnExpression) {
+					NoOp noOp = new NoOp(this, location);
+					addNode(noOp);
+
+					switchLeafNodes(node, noOp);
+					addEdge(new SequentialEdge(noOp, ret));
+				}
+			}
+		}
+		// Substitute inner RustExplicitReturn with return statements
+		else {
+			List<Statement> nonNoOpNodes = nodes.stream().filter(n -> !(n instanceof NoOp))
+					.collect(Collectors.toList());
+			if (nonNoOpNodes.size() == 1) {
+				AdjacencyMatrix<Statement, Edge, CFG> adj = getAdjacencyMatrix();
+				Statement node = nonNoOpNodes.get(0);
+				adj.getEdges().forEach(e -> adj.removeEdge(e));
+				adj.getNodes().forEach(n -> adj.removeNode(n));
+				addNode(node);
+			}
+
+			// Add return to exit node if it's the only stmt
+			if (getNodes().size() == 1) {
+				Statement onlyNode = nodes.stream().findFirst().get();
+
+				NoOp noOp = new NoOp(this, location);
+				addNode(noOp, true);
+				getEntrypoints().remove(onlyNode);
+
+				addEdge(new SequentialEdge(noOp, onlyNode));
+				getAllExitpoints().add(onlyNode);
+			}
+
+			for (Statement stmt : nodes)
+				if (stmt instanceof RustReturnExpression) {
+					Expression value = ((RustReturnExpression) stmt).getSubExpression();
+
+					Return ret = new Return(this, location, value);
+					addNode(ret);
+
+					switchLeafNodes(stmt, ret);
+				}
+		}
+
+		simplify();
+
+		Set<Edge> toRemove = new HashSet<>();
+		Set<Edge> toAdd = new HashSet<>();
+
+		AdjacencyMatrix<Statement, Edge, CFG> adj = getAdjacencyMatrix();
+		for (Edge e : adj.getEdges())
+			if (e instanceof TrueEdge) {
+				Collection<Edge> ingoinEdges = adj.getIngoingEdges(e.getDestination());
+				for (Edge ingoing : ingoinEdges) {
+					if (ingoing instanceof FalseEdge && ingoing.getSource().equals(e.getSource())) {
+						toRemove.add(e);
+						toRemove.add(ingoing);
+
+						toAdd.add(new SequentialEdge(e.getSource(), e.getDestination()));
+					}
+				}
+			}
+
+		toRemove.forEach(r -> adj.removeEdge(r));
+		toAdd.forEach(a -> adj.addEdge(a));
+	}
+
+	/**
+	 * Yields a CFG from a descriptor.
+	 * 
+	 * @param descriptor the descriptor of this CFG
+	 * @param unsafe     the decorator unsafe
+	 * @param location   the source code location where this CFG is
+	 */
+	public RustCFG(CFGDescriptor descriptor, boolean unsafe, SourceCodeLocation location) {
+		super(descriptor);
+		this.unsafe = unsafe;
+		this.location = location;
+	}
+
+	/**
+	 * Yields the decorator unsafe of this function.
+	 * 
+	 * @return true if the function is unsafe
+	 */
+	public boolean getUnsafe() {
```

Comment:
Rename in `isUnsafe`

Reply:
Fixed in commit 6962608144c610e50d0a1616902115f406f1dc57